code/game/machinery/ecto_sniffer.dm

```diff
@@ -0,0 +1,60 @@
+/obj/machinery/ecto_sniffer
+	name = "ectoscopic sniffer"
+	desc = "A highly sensitive parascientific instrument calibrated to detect the slightest whiff of ectoplasm."
+	icon = 'icons/obj/machines/research.dmi'
+	icon_state = "ecto_sniffer"
+	density = FALSE
+	anchored = FALSE
+	wires = /datum/wires/ecto_sniffer
+	///determines if the device if the power switch is turned on or off. Useful if the ghosts are too annoying.
+	var/on = TRUE
+	///If this var set to false the ghosts will not be able interact with the machine, say if the machine is silently disabled by cutting the internal wire.
+	var/sensor_enabled = TRUE
+	///List of ghost who have recently activated the device, ghosts on this list are prohibited from activating the device untill their residue decays.
+	var/list/ghosts_sampled = list()
```

Comment:
Clear this on Destroy.

Reply:
Also, these should be weakrefs so a delete of the ghost doesn't hard-del.

Reply:
I have nulled the list on destroy.

I'd love to do the weakref part but i don't know how. Maybe i can figure it out tomorrow.

Reply:
yeah this really should be weakrefs